### ISSUE TITLE
False-positive `F821: Undefined name` with `mypy_extensions.DefaultNamedArg`

### ISSUE THREAD

**Issue by actionless** [charliermarsh/ruff#1303]
```python
from typing import Any, Callable

from mypy_extensions import DefaultNamedArg

SomeCallableT = Callable[[Any, DefaultNamedArg(bool | None, name="some_prop_name")], None]
```

```console
$ ruff test_ruff_F821.py
Found 1 error(s).
test_ruff_F821.py:5:66: F821 Undefined name `some_prop_name`
```

**Comment by charliermarsh** [comment]
Good stuff, thanks for filing, an easy fix.
